internal/collector/otel_collector_plugin.go

```diff
@@ -233,7 +240,49 @@ func (oc *Collector) handleNginxConfigUpdate(ctx context.Context, msg *bus.Messa
 	}
 }
 
+func (oc *Collector) handleResourceUpdate(ctx context.Context, msg *bus.Message) {
+	resourceUpdateContext, ok := msg.Data.(*v1.Resource)
+	if !ok {
+		slog.ErrorContext(ctx, "Unable to cast message payload to *v1.Resource", "payload", msg.Data)
+		return
+	}
+
+	if oc.config.Collector.Processors.Attribute == nil {
+		oc.config.Collector.Processors.Attribute = &config.Attribute{
+			Actions: make([]config.Action, 0),
+		}
+	}
+
+	var reloadCollector bool
+	if oc.config.Collector.Processors.Attribute != nil {
+		if resourceUpdateContext.GetResourceId() != "" {
```

Comment:
Could these two be in one if statement ? 
```suggestion
		if oc.config.Collector.Processors.Attribute != nil && resourceUpdateContext.GetResourceId() != "" {
```

Reply:
Done!